code/modules/core_implant/cruciform/machinery/obelisk.dm

```diff
@@ -38,15 +38,14 @@
 	update_icon()
 	if(!active)
 		use_power = 1
-		for(var/obj/structure/burrow/burrow in range(area_radius, src))
-			if(burrow.obelisk_around == any2ref(src))
-				burrow.obelisk_around = null
 	else
 		use_power = 2
 
 		for(var/obj/structure/burrow/burrow in range(area_radius, src))
-			if(!burrow.obelisk_around)
-				burrow.obelisk_around = any2ref(src)
+			var/datum/effect/effect/system/spark_spread/sparks = new /datum/effect/effect/system/spark_spread()
```

Comment:
Don't put this in Process(), such practices destroy codebase performance.
Furthermore, people can just go around re-building obelisks around burrows to remove them. 
Make this a check in burrow's New() or initialize() or in any way before/during/after burrow's creation, don't check for them every tick.